torch_frame/data/multi_nested_tensor.py

```diff
@@ -275,6 +275,27 @@ def _single_index_select(self, index: int, dim: int) -> MultiNestedTensor:
         else:
             raise RuntimeError(f"Unsupported dim={dim} for index_select.")
 
+    def fillna_col(
+        self,
+        index: int,
+        fill_value: Union[int, float, Tensor],
+    ) -> None:
+        start_idx = torch.arange(
+            index,
+            self.num_rows * self.num_cols,
+            self.num_cols,
+            device=self.device,
+        )
+        diff = self.offset[start_idx + 1] - self.offset[start_idx]
+        batch, arange = _batched_arange(diff)
+        # Compute values
+        values = self.values[self.offset[start_idx][batch] + arange]
+        if self.values.is_floating_point():
+            values[torch.isnan(values)] = fill_value
+        else:
+            values[values == -1] = fill_value
+        self.values[self.offset[start_idx][batch] + arange] = values
```

Comment:
Is this necessary? the above in-place replacement is enough?

Reply:
Yes. We are creating a new tensor when we select some indices from a tensor. But when we are slicing, we are only creating a view.